src/libostree/ostree-sysroot-deploy.c

```diff
@@ -3589,6 +3589,49 @@ ostree_sysroot_deployment_set_kargs (OstreeSysroot     *self,
   return TRUE;
 }
 
+/**
+ * ostree_sysroot_deployment_set_kargs_in_place:
+ * @self: Sysroot
+ * @deployment: A deployment
+ * @kargs_str: (allow none): Replace @deployment's kernel arguments
+ * @cancellable: Cancellable
+ * @error: Error
+ *
+ * Replace the kernel arguments of @deployment with the values in @kargs_str.
+ */
+gboolean
+ostree_sysroot_deployment_set_kargs_in_place (OstreeSysroot     *self,
+                                              OstreeDeployment  *deployment,
+                                              char              *kargs_str,
+                                              GCancellable      *cancellable,
+                                              GError           **error)
+{
+  if (!_ostree_sysroot_ensure_writable (self, error))
+    return FALSE;
+
+  g_assert (!ostree_deployment_is_staged (deployment));
+
+  OstreeBootconfigParser *new_bootconfig = ostree_deployment_get_bootconfig (deployment);
+  ostree_bootconfig_parser_set (new_bootconfig, "options", kargs_str);
+
+  g_autofree char *bootconf_name =
+    g_strdup_printf ("ostree-%d-%s.conf",
+                     self->deployments->len - ostree_deployment_get_index (deployment),
+                     ostree_deployment_get_osname (deployment));
+
+  g_autofree char *bootconfdir = g_strdup_printf ("loader.%d/entries", self->bootversion);
```

Comment:
The bug has something to do with `self->bootversion` here I think in the unit test case.  But...it should be correct as is AFAICS...

Reply:
Add assert before https://github.com/ostreedev/ostree/blob/main/tests/test-admin-kargs.sh#L32, check `sysroot/boot/loader.1/entries` exists, can not find `loader.1/entries`, is this expected?
```
+ assert_has_dir sysroot/boot/loader.1/entries
+ test -d sysroot/boot/loader.1/entries

+ assert_has_dir loader.1/entries
+ test -d loader.1/entries
+ fatal 'Couldn'\''t find '\''loader.1/entries'\'''
```

Reply:
OK running this under `strace`, what I see is:

```
openat(AT_FDCWD, "loader.1/entries", O_RDONLY|O_NOCTTY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
```

which implies that `self->boot_fd` is `-1`.  Looking...